src/instructlab/data/generate.py

```diff
@@ -143,7 +143,7 @@
 @click.option(
     "--pipeline",
     type=click.STRING,
-    default="simple",
+    cls=clickext.ConfigOption,
     help="Data generation pipeline to use. Available: simple, full, or a valid path to a directory of pipeline workflow YAML files. Note that 'full' requires a larger teacher model, Mixtral-8x7b.",
```

Comment:
(Observation) These should probably be enforced as enum CLI option. Same for config option allowed values.

Reply:
+1 it's also pretty easy to implement. Something like this:
```
from enum import Enum

...

class SDGPipeline(Enum):
    simple = 0
    full = 1
...

pipeline = SDGPipeline(pipeline_arg)  # assuming pipeline_arg is the arg passed in the CLI in string format, for example, "simple"
# pipeline will be < SDGPipeline.simple: 0>
```

This way you can still do the same checking against CLI args while having a more robust approach overall with Enums.

Reply:
Paths to custom pipelines also need to be allowed.  Not sure if there is a way to validate enum or random str.

Reply:
This might be a good follow up PR once this is merged.